### ISSUE TITLE
Pricing Rule not applied for the child of selected Item Group

### ISSUE THREAD

**Issue by sjain17** [frappe/erpnext#32696]
### Information about bug

If we try to apply pricing on the item group tree then it is not working for child item groups.

For example - I have an Item group structured like this -
1 FMCG
1.1 GROCERY
1.2 BEVERAGES
1.1.1 FLOUR
1.1.2 FAST FOOD
1.1.3 INSTANT FOOD
1.2.1 BOTTLED WATER
1.2.2 COLD DRINKS

Now if I apply a pricing rule or promotional scheme on FMCG then it is not getting applied in GROCERY, BEVERAGES, FLOUR, FAST FOOD, INSTANT FOOD, BOTTLED WATER, and COLD DRINKS.

However, it should be applied to all child item groups as well.

### Module

selling

### Version

Frappe - Version-13
ERPNext - Version-13

### Relevant log output / Stack trace / Full Error Message.

```shell
There is no error but the output is not as per expectations.
```

**Comment by maharshivpatel** [comment]
@sjain17 Pricing rule is only applied on direct items of specified group.  Pricing rule on nested group is not supported.

**Comment by sjain17** [comment]
Yeah, that's correct. Is it going to be available in the next release, it is very difficult to apply the pricing rule on the child item group level.

**Comment by maharshivpatel** [comment]
Currently, There is no active development going on for this feature and lot of existing pricing rule setups will break if we introduce this. I will think about this you can also contribute by raising PR.

**Comment by sjain17** [comment]
Ok, we will do the development and raise a PR then.

**Comment by adityaduggal** [comment]
I have checked it in version-13 its a bug since it used to work out of the box in version-12
Also the priority is somehow not getting fetched if we have multiple item group based Pricing Rules.
Suppose you have priority 1 Pricing Rule on FMCG
Then if you create a priority 2 Pricing Rule on GROCERY then ideally priority 2 should be applied on the Grocery Item Group instead of Priority 1 but in reality its Priority 1 which is getting applied.
